### ISSUE TITLE
image pre-processing

### ISSUE THREAD

**Issue by xnholiday** [AbhishekRS4/seg_net#1]
Hello, I would like to ask, how to perform the image pre-processing ?

**Comment by AbhishekRS4** [comment]
SegNet uses VGG-16 pretrained encoder. The input image has to be in **BGR** and **the image_net_mean has to be subtracted**. This is done in the **parse_fn()**. The following code does this.

`img_r, img_g, img_b = tf.split(value=img, axis=2, num_or_size_splits=3)`
`img = tf.concat(values=[img_b, img_g, img_r], axis=2)`
`img = img - IMAGENET_MEAN`

**Comment by xnholiday** [comment]
> SegNet uses VGG-16 pretrained encoder. The input image has to be in **BGR** and **the image_net_mean has to be subtracted**. This is done in the **parse_fn()**. The following code does this.
> 
> `img_r, img_g, img_b = tf.split(value=img, axis=2, num_or_size_splits=3)`
> `img = tf.concat(values=[img_b, img_g, img_r], axis=2)`
> `img = img - IMAGENET_MEAN`

Sorry, I mean that how to resize the CityScapes dataset and input the resized image into the segnet?

**Comment by AbhishekRS4** [comment]
Resizing of images, labels was done beforehand when I trained. If you want to do it after reading the images, take a look at https://www.tensorflow.org/api_docs/python/tf/image/resize_images/

**Comment by xnholiday** [comment]
> Resizing of images, labels was done beforehand when I trained. If you want to do it after reading the images, take a look at https://www.tensorflow.org/api_docs/python/tf/image/resize_images/

Firstly, I feel so sorry that I reply your email so late for some reasons. And thanks a lot, I will read it carefully.
